diff --git a/core.c b/core.c
index 0687cc2..d1eaa0c 100644
--- a/core.c
+++ b/core.c
@@ -544,9 +544,6 @@ static void tps6598x_handle_plug_event(struct tps6598x *tps, u32 status)
 static void cd321x_typec_update_mode(struct tps6598x *tps, struct cd321x_status *st)
 {
 	struct cd321x *cd321x = container_of(tps, struct cd321x, tps);
-	struct typec_thunderbolt_switch_data tbt_switch_data = {
-		.state = TYPEC_THUNDERBOLT_SWITCH_OFF,
-	};
 
 	if (!(st->data_status & TPS_DATA_STATUS_DATA_CONNECTION)) {
 		if (cd321x->state.mode == TYPEC_STATE_SAFE)
@@ -554,7 +551,6 @@ static void cd321x_typec_update_mode(struct tps6598x *tps, struct cd321x_status
 		cd321x->state.alt = NULL;
 		cd321x->state.mode = TYPEC_STATE_SAFE;
 		cd321x->state.data = NULL;
-		typec_thunderbolt_switch_set(cd321x->tbt_switch, &tbt_switch_data);
 		typec_mux_set(cd321x->mux, &cd321x->state);
 	} else if (st->data_status & TPS_DATA_STATUS_DP_CONNECTION) {
 		struct typec_displayport_data dp_data;
@@ -594,7 +590,6 @@ static void cd321x_typec_update_mode(struct tps6598x *tps, struct cd321x_status
 		cd321x->state.alt = cd321x->port_altmode_dp;
 		cd321x->state.data = &dp_data;
 		cd321x->state.mode = mode;
-		typec_thunderbolt_switch_set(cd321x->tbt_switch, &tbt_switch_data);
 		typec_mux_set(cd321x->mux, &cd321x->state);
 	} else if (st->data_status & TPS_DATA_STATUS_TBT_CONNECTION) {
 		struct typec_thunderbolt_data tbt_data;
@@ -620,13 +615,6 @@ static void cd321x_typec_update_mode(struct tps6598x *tps, struct cd321x_status
 		cd321x->state.mode = TYPEC_TBT_MODE;
 		cd321x->state.data = &tbt_data;
 		typec_mux_set(cd321x->mux, &cd321x->state);
-
-		tbt_switch_data.state = TYPEC_THUNDERBOLT_SWITCH_TBT;
-		tbt_switch_data.tbt = tbt_data;
-		tbt_switch_data.orientation = TPS_STATUS_TO_UPSIDE_DOWN(st->status) ?
-						      TYPEC_ORIENTATION_REVERSE :
-						      TYPEC_ORIENTATION_NORMAL;
-		typec_thunderbolt_switch_set(cd321x->tbt_switch, &tbt_switch_data);
 	} else if (st->data_status & CD321X_DATA_STATUS_USB4_CONNECTION) {
 		struct enter_usb_data eusb_data;
 
@@ -641,20 +629,12 @@ static void cd321x_typec_update_mode(struct tps6598x *tps, struct cd321x_status
 		cd321x->state.data = &eusb_data;
 		cd321x->state.mode = TYPEC_MODE_USB4;
 		typec_mux_set(cd321x->mux, &cd321x->state);
-
-		tbt_switch_data.state = TYPEC_THUNDERBOLT_SWITCH_USB4;
-		tbt_switch_data.usb4 = eusb_data;
-		tbt_switch_data.orientation = TPS_STATUS_TO_UPSIDE_DOWN(st->status) ?
-						      TYPEC_ORIENTATION_REVERSE :
-						      TYPEC_ORIENTATION_NORMAL;
-		typec_thunderbolt_switch_set(cd321x->tbt_switch, &tbt_switch_data);
 	} else {
 		if (cd321x->state.alt == NULL && cd321x->state.mode == TYPEC_STATE_USB)
 			return;
 		cd321x->state.alt = NULL;
 		cd321x->state.mode = TYPEC_STATE_USB;
 		cd321x->state.data = NULL;
-		typec_thunderbolt_switch_set(cd321x->tbt_switch, &tbt_switch_data);
 		typec_mux_set(cd321x->mux, &cd321x->state);
 	}
 
@@ -668,9 +648,6 @@ static void cd321x_update_work(struct work_struct *work)
 					    struct cd321x, update_work);
 	struct tps6598x *tps = &cd321x->tps;
 	struct cd321x_status st;
-	struct typec_thunderbolt_switch_data tbt_switch_data = {
-		.state = TYPEC_THUNDERBOLT_SWITCH_OFF,
-	};
 
 	guard(mutex)(&tps->lock);
 
@@ -721,7 +698,6 @@ static void cd321x_update_work(struct work_struct *work)
 
 	/* If there was a disconnection, set PHY to off */
 	if (!new_connected || was_disconnected) {
-		typec_thunderbolt_switch_set(cd321x->tbt_switch, &tbt_switch_data);
 		cd321x->state.alt = NULL;
 		cd321x->state.mode = TYPEC_STATE_SAFE;
 		cd321x->state.data = NULL;
@@ -1221,12 +1197,6 @@ cd321x_register_port(struct tps6598x *tps, struct fwnode_handle *fwnode)
 		goto err_unregister_altmodes;
 	}
 
-	cd321x->tbt_switch = fwnode_typec_thunderbolt_switch_get(fwnode);
-	if (IS_ERR(cd321x->tbt_switch)) {
-		ret = PTR_ERR(cd321x->tbt_switch);
-		goto err_unregister_mux;
-	}
-
 	cd321x->state.alt = NULL;
 	cd321x->state.mode = TYPEC_STATE_SAFE;
 	cd321x->state.data = NULL;
@@ -1234,9 +1204,6 @@ cd321x_register_port(struct tps6598x *tps, struct fwnode_handle *fwnode)
 
 	return 0;
 
-err_unregister_mux:
-	typec_mux_put(cd321x->mux);
-	cd321x->mux = NULL;
 err_unregister_altmodes:
 	typec_unregister_altmode(cd321x->port_altmode_dp);
 	typec_unregister_altmode(cd321x->port_altmode_tbt);
@@ -1258,8 +1225,6 @@ cd321x_unregister_port(struct tps6598x *tps)
 {
 	struct cd321x *cd321x = container_of(tps, struct cd321x, tps);
 
-	typec_thunderbolt_switch_put(cd321x->tbt_switch);
-	cd321x->tbt_switch = NULL;
 	typec_mux_put(cd321x->mux);
 	cd321x->mux = NULL;
 	typec_unregister_altmode(cd321x->port_altmode_dp);
diff --git a/tps6598x.h b/tps6598x.h
index 6cf6694..9c4bca9 100644
--- a/tps6598x.h
+++ b/tps6598x.h
@@ -377,8 +377,6 @@ struct cd321x {
 	struct typec_mux *mux;
 	struct typec_mux_state state;
 
-	struct typec_thunderbolt_switch *tbt_switch;
-
 	struct cd321x_status update_status;
 	struct delayed_work update_work;
 	struct usb_pd_identity cur_partner_identity;
